### ISSUE TITLE
How does WebPushChannel::deleteInvalidSubscriptions work?

### ISSUE THREAD

**Issue by kyonkopa** [laravel-notification-channels/webpush#39]
What's the criteria for invalidity, as subscriptions are deleted simply because something went, at least that what is happening in my case, because after using the same subscription data a second or third time it works.

**Comment by cretueusebiu** [comment]
See https://github.com/web-push-libs/web-push-php/blob/master/README.md#server-errors

**Comment by kyonkopa** [comment]
I have figured out what it happening, `WebPushChannel::deleteInvalidSubscriptions` is looping through the **error response** and for each subscription the push message was sent its going to delete the subscription.
Now this happens **if there was an error** not because the subscription is 'invalid'.

Inspecting the error returned shows
> cURL error 6: Could not resolve host: xxxxxxxxxxxxxxxxxxxxxxxx
in the message key, suggesting internet connection problem, I got this error when I had disconnected my PC to the internet.

So I think what should happen is if there was an error while sending a push message, the error response should be examined before the subscriptions are deleted.
